pkg/controllers/fpga/controller.go

```diff
@@ -65,6 +66,27 @@ func (c *controller) CreateEmptyObject() client.Object {
 	return &devicepluginv1.FpgaDevicePlugin{}
 }
 
+func (c *controller) Upgrade(ctx context.Context, obj client.Object) (upgrade bool) {
```

Comment:
This function seem to be quite generic. Can we avoid code duplication if we move the code into the function that accepts an interface with a couple of methods like GetImage and GetInitImage?

Reply:
@bart0sh 
In principle this could be done, but we would need 4 methods in each device controller, i.e. GetImage, GetInitImage, SetImage, SetInitImage. Should we go for it?